### ISSUE TITLE
Test case fails after adding qute dependency

### ISSUE THREAD

**Issue by faskan** [quarkusio/quarkus#24312]
### Describe the bug

I am working on a Quarkus AWS lambda service that generates some html reports and uploads to S3. Unfortunately my test cases started failing (with a 404) after adding quarkus-resteasy-qute dependency.

I have replicated the same in a starter project (with AWS lambda) [here](https://github.com/faskan/quarkus-lambda-qute/). 

### Expected behavior

The test case should not fail when adding quarkus-resteasy-qute dependency

### Actual behavior

Test case fails with a 404

### How to Reproduce?

1. Generate a quarkus starter project with AWS lambda
2. Run the test case and everything works fine
3. Add quarkus-resteasy-qute dependency
4. Test case fails with 404

### Output of `uname -a` or `ver`

Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:54 PST 2021; root:xnu-8019.61.5~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk 11.0.11 2021-04-20

### Quarkus version or git rev

2.7.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4

**Comment by mkouba** [comment]
I don't think that you can just add the `quarkus-resteasy-qute` dependency or `quarkus-resteasy` and expect it to work OOTB. But I'm definitely not an amazon lambda expert. CC @geoand @patriot1burke

**Comment by geoand** [comment]
Neither am I, but I also assume it won't work

**Comment by faskan** [comment]
I am not expecting it to work OOB. In my original project, I have a Template injected in to a service class and it generates reports and saves to S3. But as soon as I add this dependency, my test case starts failing.

**Comment by patriot1burke** [comment]
quarkus-amazon-lambda-http (-rest) + resteasy + qute should work OOTB. (Doesn't resteasy + qute work OOTB?)

I'll try to replicate.

**Comment by patriot1burke** [comment]
Sorry I didn't notice this before, but your project includes quarkus-resteasy-qute which will *NOT* work with regular lambda.  I was able to run your project by including quarkus-qute (without resteasy).  It would work with quarkus-amazon-lambda-http (or rest) which spins up an API gateway on AWS though.

Hope I'm making sense, I'm going to close this bug.
